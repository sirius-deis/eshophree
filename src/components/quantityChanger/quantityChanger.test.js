import { render, screen, fireEvent } from "@testing-library/react";
import QuantityChanger from "./quantityChanger";

describe("QuantityChanger", () => {
  const mockChangeFn = jest.fn();
  beforeEach(() => {
    mockChangeFn.mockClear();
  })
  it("should match snapshot", () => {
    const { container } = render(<QuantityChanger quantity={4} changeFn={mockChangeFn}/>);
    expect(container).toMatchSnapshot();
  })
  it("should render the quantity and buttons", () => {
    render(<QuantityChanger quantity={4} changeFn={mockChangeFn}/>);
    expect(screen.getByText("4")).toBeInTheDocument();
    expect(screen.getByText("+")).toBeInTheDocument();
    expect(screen.getByText("-")).toBeInTheDocument();
  })
  it("should call changeFn with 1 when + button is clicked", () => {
    render(<QuantityChanger quantity={4} changeFn={mockChangeFn}/>);
    fireEvent.click(screen.getByText("+"));
    expect(mockChangeFn).toHaveBeenCalledWith(1);
  })
  it("should call changeFn with -1 when - button is clicked", () => {
    render(<QuantityChanger quantity={4} changeFn={mockChangeFn}/>);
    fireEvent.click(screen.getByText("-"));
    expect(mockChangeFn).toHaveBeenCalledWith(-1);
  })
  it('renders with different quantities', () => {
    const {rerender} = render(<QuantityChanger quantity={4} changeFn={mockChangeFn} />);
    expect(screen.getByText("4")).toBeInTheDocument();

    rerender(<QuantityChanger quantity={7} changeFn={mockChangeFn} />);
    expect(screen.getByText("7")).toBeInTheDocument();
  })
})