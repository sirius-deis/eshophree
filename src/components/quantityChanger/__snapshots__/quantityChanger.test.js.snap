// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuantityChanger should match snapshot 1`] = `
<div>
  <div>
    <button
      class="sc-bgqQcB sc-gTRrQi jHWwsb cVPPIc"
      type="button"
    >
      +
    </button>
    4
    <button
      class="sc-bgqQcB sc-gTRrQi jHWwsb cVPPIc"
      type="button"
    >
      -
    </button>
  </div>
</div>
`;
